### PR TITLE
feat: Add --gitignore flag to read in .gitignore files

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Updates `@eslint/migrate-config` to use the new `gitignore` API added in https://github.com/eslint/rewrite/pull/47

#### What changes did you make? (Give an overview)

- When `--gitignore` is passed as an argument, a code block is added to use `includeIgnoreFile()`.
- Updated README
- Added tests  

#### Related Issues

refs #44

### CODE REVIEW

**Review by mdjermanovic — COMMENTED** [review]
I tried this with a simple `.eslintrc.json` example:

```json
{
    "rules": {
        "no-unused-vars": 2
    }
}
```

`eslint.config.cjs` result:

```js
const {
    includeIgnoreFile,
} = require("@eslint/compat");

const gitignorePath = path.resolve(__dirname, ".gitignore");

module.exports = [includeIgnoreFile(gitignorePath), {
    rules: {
        "no-unused-vars": 2,
    },
}];
```

`eslint.config.mjs` result:

```js
import { includeIgnoreFile } from "@eslint/compat";
const gitignorePath = path.resolve(__dirname, ".gitignore");

export default [includeIgnoreFile(gitignorePath), {
    rules: {
        "no-unused-vars": 2,
    },
}];
```

In the cjs config, importing `path` is missing. In the mjs config, importing `path` and calculating `__dirname` are missing.

**Comment by nzakas** [comment]
Ah good catch! I'll fix that.

File: packages/migrate-config/tests/fixtures/gitignore-simple/expected.mjs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { includeIgnoreFile } from "@eslint/compat";
+import path from "node:path";
+
+const __filename = fileURLToPath(import.meta.url);
```
Comment (mdjermanovic) [inline packages/migrate-config/tests/fixtures/gitignore-simple/expected.mjs:4]: Missing import of `fileURLToPath`.

File: packages/migrate-config/src/migrate-config.js [hunk 2]
```diff
@@ -839,15 +892,22 @@ export function migrateConfig(config, { sourceType = "module" } = {}) {
...
 		});
-		migration.inits.push(...getFlatCompatInit(sourceType));
+		migration.needsDirname ||= isModule;
+		migration.inits.push(...getFlatCompatInit());
+	}
+
+	// add .gitignore if necessary
+	if (gitignore) {
+		migration.needsDirname ||= isModule;
+		configArrayElements.unshift(createGitignoreEntry(migration));
```
Comment (mdjermanovic) [inline packages/migrate-config/src/migrate-config.js:902]: There's a type error here:

```
Argument of type 'CallExpression' is not assignable to parameter of type 'ObjectExpression | SpreadElement'.
  Property 'argument' is missing in type 'CallExpression' but required in type 'SpreadElement'.
```

Comment (nzakas) [reply packages/migrate-config/src/migrate-config.js:902]: Actually fixed this and forgot to push.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**